src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php

```diff
@@ -139,6 +148,32 @@ protected function tokenExpired($createdAt)
         return Carbon::parse($createdAt)->addSeconds($this->expires)->isPast();
     }
 
+    /**
+     * Determine if a token record exists and was recently created.
+     *
+     * @param \Illuminate\Contracts\Auth\CanResetPassword $user
+     * @return bool
+     */
+    public function recentlyCreated(CanResetPasswordContract $user)
+    {
+        $record = (array) $this->getTable()->where(
```

Comment:
And return true here if `$timeout === null` would be a better idea so:
- it doesn't makes unnecessary db queries
- it will not return the error link when the request thinks it's happened in the same second (which shouldn't be but who knows)

Reply:
If these two corrections are really needed for 6.x then IMHO it's better to make this fix only in 7.x, isn't it?

Reply:
This depends, the first question should be a user be able to disable this feature?

If yes then the checks should be included by default, maybe even with `$this->timeout <= 0` instead of the null check. In this case the value can stay 0 in the config and I would set it to 0 for the class to.

If a new installation is used it will include it in the config, the old ones need the manual enabling in either way. L7 can set sensible default value (60).